File: KahootClient/src/RegisterModel.ts
export class RegisterModel{
  public id: number | undefined;

  constructor(public userName: string | undefined,
              public password: string | undefined,
              public confirmPassword: string | undefined,
              public email: string | undefined,
              public birthday: Date | undefined, ) {  }

  public get Id() {
    return this.id;
  }

  public get Username() {
    return this.userName;
  }

  public get Password() {
    return this.password;
  }

  public get ConfirmPassword() {
    return this.confirmPassword;
  }

  public get Email() {
    return this.email;
  }

  public get Birthday() {
    return this.birthday;
  }
};

File: KahootWebApi/ViewModels/RegisterModel.cs
using System.ComponentModel.DataAnnotations;

namespace KahootWebApi.ViewModels
{
    public class RegisterModel
    {
        [Required]
        public string? UserName { get; set; }

        [Required]
        [DataType(DataType.Password)]
        public string? Password { get; set; }

        [DataType(DataType.Password)]
        [Compare("Password")]
        public string? ConfirmPassword { get; set; }

        [Required]
        [DataType(DataType.EmailAddress)]
        public string? Email { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime? Birthday { get; set; }

        //[Required]
        //[DataType(DataType.Text)]
        //public string? Role { get; set; }
    }
}
